src/WebJobs.Script/Description/PowerShell/PowerShellFunctionInvoker.cs

```diff
@@ -0,0 +1,451 @@
+                    PowerShellConstants.AdditionChar, PowerShellConstants.FullyQualifiedErrorIdLabel,
+                    errorRecord.FullyQualifiedErrorId));
+            }
+
+            return stringBuilder.ToString();
+        }
+
+        internal static string GetStackTrace(string functionName, string scriptStackTrace, string fileName)
+        {
+            string stackTrace = scriptStackTrace.Replace(PowerShellConstants.StackTraceScriptBlock, fileName);
+
+            if (stackTrace.Contains(functionName))
+            {
+                string[] tokens = stackTrace.Split((char[])null, StringSplitOptions.RemoveEmptyEntries);
+                string[] newtokens = new string[tokens.Length];
+                int index = 0;
+                foreach (string token in tokens)
+                {
+                    if (token.Contains(functionName))
+                    {
+                        string relativePath = GetRelativePath(functionName, token);
+                        newtokens[index++] = relativePath;
+                    }
+                    else if (token.Contains(PowerShellConstants.StackTraceScriptBlock))
+                    {
+                        newtokens[index++] = token.Replace(PowerShellConstants.StackTraceScriptBlock, fileName);
+                    }
+                    else
+                    {
+                        newtokens[index++] = token;
+                    }
+                }
+
+                stackTrace = string.Join(" ", newtokens);
+            }
+
+            return stackTrace;
+        }
+
+        internal static string GetScript(string scriptFilePath)
+        {
+            string script = null;
+            if (File.Exists(scriptFilePath))
+            {
+                script = File.ReadAllText(scriptFilePath);
+            }
+
+            return script;
+        }
+
+        internal static List<string> GetModuleFilePaths(string rootScriptPath, string functionName)
+        {
+            List<string> modulePaths = new List<string>();
+            string functionFolder = Path.Combine(rootScriptPath, functionName);
+            string moduleDirectory = Path.Combine(functionFolder, PowerShellConstants.ModulesFolderName);
+            if (Directory.Exists(moduleDirectory))
+            {
+                modulePaths.AddRange(Directory.GetFiles(moduleDirectory,
+                    PowerShellConstants.ModulesManifestFileExtensionPattern));
+                modulePaths.AddRange(Directory.GetFiles(moduleDirectory,
+                    PowerShellConstants.ModulesBinaryFileExtensionPattern));
+                modulePaths.AddRange(Directory.GetFiles(moduleDirectory,
+                    PowerShellConstants.ModulesScriptFileExtensionPattern));
+            }
+
+            return modulePaths;
+        }
+
+        private void InitializeEnvironmentVariables(Dictionary<string, string> environmentVariables, string functionInstanceOutputPath, object input, ExecutionContext context)
```

Comment:
I believe this function is a complete unmodified duplicate of ScriptFunctionInvoker.InitializeEnvironmentVariables. Please add a new ScriptFunctionInvoker base class (that itself derives from FunctionInvoker), and add this as a protected method. That way your code and ScriptFunctionInvoker can call it.